Lib/test/test_capi/test_getargs.py

```diff
@@ -1348,6 +1348,123 @@ def test_nested_tuple(self):
                     "argument 1 must be sequence of length 1, not 0"):
                 parse(((),), {}, '(' + f + ')', ['a'])
 
+    def test_specific_type_errors(self):
+        parse = _testcapi.parse_tuple_and_keywords
+
+        def check(format, arg, expected, got='list'):
+            errmsg = f'must be {expected}, not {got}'
+            with self.assertRaisesRegex(TypeError, errmsg):
+                parse((arg,), {}, format, ['a'])
+
+        check('k', [], 'int')
```

Comment:
Maybe you want to add a test for `'?'` only, without anything (and see that it should be rejected). Or maybe `?` could indicate `None or nothing`?

Reply:
This is indirectly tested in test_skipitem.